### PR TITLE
[JENKINS-26107] Examples for blocky stage

### PR DESCRIPTION
**Pull request by jglick**
For after https://github.com/jenkinsci/pipeline-stage-step-plugin/pull/5 is released.

@reviewbybees esp. @kohsuke

### CODE REVIEW

File: src/main/js/samples.js [hunk 1]
```diff
@@ -46,31 +46,25 @@ samples.push({
     name: 'hello',
     title: 'Hello World',
     script: "node {\n" +
-        "   stage 'Stage 1'\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: Just seemed gratuitous here.

File: src/main/webapp/snippets/workflow.js [hunk 2]
```diff
@@ -62,35 +62,24 @@ snippet for++\n\
...
-# node\n\
-snippet node\n\
-	node {\n\
-	    ${1}\n\
-	}\n\
-# node-l ('label') \n\
-snippet node-l\n\
-	node ('${1}') {\n\
-	    \n\
-	}\n\
```
Comment (jglick) [inline src/main/webapp/snippets/workflow.js]: redundant

File: src/main/webapp/snippets/workflow.js [hunk 3]
```diff
@@ -62,35 +62,24 @@ snippet for++\n\
...
+	}\n\
 # node\n\
 snippet node\n\
 	node {\n\
 	    ${1}\n\
 	}\n\
 # node-l ('label') \n\
 snippet node-l\n\
-	node ('${1}') {\n\
+	node('${1}') {\n\
```
Comment (jglick) [inline src/main/webapp/snippets/workflow.js]: Bringing snippet style in line with what `Snippetizer` and most docs would suggest.

**Comment by kohsuke** [comment]
:bee:

File: src/main/js/samples.js [hunk 4]
```diff
@@ -45,38 +45,37 @@ function getSample(sampleName) {
 samples.push({
     name: 'hello',
     title: 'Hello World',
-    script: "node {\n" +
-        "   stage 'Stage 1'\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: Just seemed gratuitous here.

File: src/main/js/samples.js [hunk 5]
```diff
@@ -45,38 +45,37 @@ function getSample(sampleName) {
...
+        "   }\n" +
+        "   stage('Build') {\n" +
+        "      // Run the maven build\n" +
+        "      if (isUnix()) {\n" +
+        "         sh \"'${mvnHome}/bin/mvn' -Dmaven.test.failure.ignore clean package\"\n" +
+        "      } else {\n" +
+        "         bat(/\"${mvnHome}\\bin\\mvn\" -Dmaven.test.failure.ignore clean package/)\n" +
+        "      }\n" +
+        "   }\n" +
+        "   stage('Results') {\n" +
```
Comment (jglick) [inline src/main/js/samples.js]: Adding a new stage here.

**Comment by svanoort** [comment]
@jglick Block scoped stage is now live and hitting the update center, can we go ahead and merge?

**Comment by jglick** [comment]
Yup, let me just do one last sanity check…

File: pom.xml [hunk 6]
```diff
@@ -153,12 +153,6 @@
         </dependency>
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>pipeline-stage-step</artifactId>
```
Comment (svanoort) [inline pom.xml]: 🐛  you'll still need a dependency on pipeline-stage-step 2.2 for the example to actually run.

Comment (jglick) [inline pom.xml]: Actually running the example requires all sorts of plugins which `workflow-cps` does not depend on. If we could define this snippets in another plugin like `workflow-aggregator`, I would do that, but @tfennelly did not write such an extension point.

Comment (jglick) [inline pom.xml]: Anyway this was a `test`-scoped dependency to begin with, which was only needed by the previous version of `SnippetizerTest`.

Comment (svanoort) [inline pom.xml]: Ah, okay that makes it clearer.  Okay, retracting my bug and granting the 🐝 instead!

**Pull request merged**